multicluster/controllers/multicluster/leader_clusterset_controller.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2021 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multicluster
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	multiclusterv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"antrea.io/antrea/multicluster/controllers/multicluster/core"
+)
+
+// LeaderClusterSetReconciler reconciles a ClusterSet object
+// There will be one controller running in each namespace of the leader for multiple cluster-set support
+// so each controller will only be handling a single cluster set in the given namespace.
+// TODO: Split the reconciler for member and leader to avoid if checks. In case a cluster is both a
```

Comment:
remove this TODO?

Reply:
Yeah I think it is removed in the latest commit